test/suites/RentingDappManager/RentingOperation.test.ts

```diff
@@ -0,0 +1,668 @@
+            const contractLogs = parseLogs1.map(ele => ele.args);
+
+            const rentalContract = contractLogs[0][2];
+
+            let rentalAgreementESN_1 = RentalAgreementFactory.connect(
+                rentalContract,
+                wallet1 ?? global.providerESN
+            );
+            let rentalAgreementESN_2 = RentalAgreementFactory.connect(
+                rentalContract,
+                wallet2 ?? global.providerESN
+            );
+            
+
+            // Initial check by lessor
+            const receipt1 = await parseReceipt(
+                rentalAgreementESN_1
+                    .connect(wallet1.connect(global.providerESN))
+                    .initialCheckByLessor(1),
+                // true
+            );
+            const parsedLogs1 = receipt1.logs.map((log) =>
+                rentalAgreementESN_1.interface.parseLog(log)
+            );
+            ok(
+                parsedLogs1[0].args[0] === 0,
+                "Check should be lessor_confirmed for rent"
+            );
+            
+
+            // Initial check by lessee
+            await global.providerESN.getSigner(0).sendTransaction({
+                to: wallet2.address,
+                value: parseEther('10000'),
+            });
+            const receipt2 = await parseReceipt(
+                rentalAgreementESN_2
+                    .connect(wallet2.connect(global.providerESN))
+                    .initialCheckByLessee( 1, {value: (contractLogs[0][6])} ),
+                //true
+            );
+            const parsedLogs2 = receipt2.logs.map((log) => 
+                rentalAgreementESN_2.interface.parseLog(log)
+            )
+            ok(
+                parsedLogs2[0].args[0] === 1,
+                "Check should be lessee_confirmed for rent"
+            );
+
+
+            // Pay rent for once
+            const receipt3 = await parseReceipt(
+                rentalAgreementESN_2
+                    .connect(wallet2.connect(global.providerESN))
+                    .payRent({value: contractLogs[0][5]}),
+                //true
+            );
+            const parsedLogs3 = receipt3.logs.map((log) =>
+                rentalAgreementESN_2.interface.parseLog(log)
+            );
+            ok(
+                formatEther(parsedLogs3[0].args[0]) === formatEther(contractLogs[0][5]),
+                "Amount of rent paid is not correct"
+            );
+
+
+            // Final check by lessor
+            const receipt4 = await parseReceipt(
+                rentalAgreementESN_1
+                    .connect(wallet1.connect(global.providerESN))
+                    .finalCheckByLessor(1),
+                // true
+            );
+            const parsedLogs4 = receipt4.logs.map((log) =>
+                rentalAgreementESN_1.interface.parseLog(log)
+            );
+            ok(
+                parsedLogs4[0].args[0] === 3,
```

Comment:
If you need to check equality, you can instead use `strictlyEqual` instead of `ok`.

Reply:
Here's an [example](https://github.com/KMPARDS/esn-contracts/blob/master/test/suites/BuildSurvey/AddSurvey.test.ts#L77).

Reply:
Done in equality check areas